agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/CustomClassWriterv2.java

```diff
@@ -0,0 +1,138 @@
+package com.microsoft.applicationinsights.agent.internal.agent;
+
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Opcodes;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This class overwrites default class writer of ASM to use the ClassLoader
+ * provided by DefaultByteCode transformer (This loader essentially has all
+ * the required classes already loaded)
+ */
+public class CustomClassWriterv2 extends org.objectweb.asm.ClassWriter {
```

Comment:
I'm not a fan of _SomeClass**V2**_ naming schemes. If the modifications can't be described in one or two words, the go with "branding" it; e.g. _AppInsightsCustomClassWriter_.

Reply:
I think this can be just refactored as CustomClassWriter I will do that.